src/main/java/ch/ivyteam/ivy/maven/AbstractEngineMojo.java

```diff
@@ -129,9 +142,36 @@ protected final File findMatchingEngineInCacheDirectory() throws MojoExecutionEx
         versionOfEngineToTake = candidateVersion;
       }
     }
+    
+    if (engineToTarget && project != null && engineDirToTake != null)
+    {
+      engineDirToTake = copyEngineToTarget(engineDirToTake);
+    }
+    
     return engineDirToTake;
   }
 
+  private File copyEngineToTarget(File engineDirToTake)
```

Comment:
how it the intended behaviour in multi-module builds? will the same engine be used for all projects? or one per project? .

Reply:
This implementation would copy the engine for each project, sharing is not possible.